### PR TITLE
Fix managing coordinators in HA

### PR DESCRIPTION
**Pull request by as51340**
### Description

Coordinator aren't explicitly saved in raft log. They are auto-managed by using `srv_config` from NuRaft. Aux field is used for saving more info about coordinators.

[master < Task] PR
- [x] Provide the full content or a guide for the final git message
    - **Fix managing coordinators in HA**

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[Release note text]**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/coordination/raft_state.cpp [hunk 1]
```diff
@@ -162,6 +165,28 @@ auto RaftState::AddCoordinatorInstance(coordination::CoordinatorToCoordinatorCon
   }
 }
 
+auto RaftState::GetCoordinatorInstances() const -> std::vector<CoordinatorToCoordinatorConfig> {
+  std::vector<ptr<srv_config>> srv_configs;
+  raft_server_->get_srv_config_all(srv_configs);
+
+  auto aux1 = raft_server_->get_aux(0);
+  auto aux2 = raft_server_->get_aux(2);
+  auto aux3 = raft_server_->get_aux(2);
```
Comment (antoniofilipovic) [inline src/coordination/raft_state.cpp:174]: Why get_aux(2) two times?

File: src/coordination/raft_state.cpp [hunk 2]
```diff
@@ -126,11 +127,13 @@ auto RaftState::InstanceName() const -> std::string { return fmt::format("coordi
...
 auto RaftState::RaftSocketAddress() const -> std::string { return raft_endpoint_.SocketAddress(); }
 
-auto RaftState::AddCoordinatorInstance(coordination::CoordinatorToCoordinatorConfig const &config) -> void {
+auto RaftState::AddCoordinatorInstance(CoordinatorToCoordinatorConfig const &config) -> void {
   spdlog::trace("Adding coordinator instance {} start in RaftState for coordinator_{}", config.coordinator_id,
                 coordinator_id_);
   auto const endpoint = config.coordinator_server.SocketAddress();
-  srv_config const srv_config_to_add(static_cast<int>(config.coordinator_id), endpoint);
+  auto const aux = nlohmann::json(config).dump();
+  spdlog::trace("Aux data: {}", aux);
```
Comment (antoniofilipovic) [inline src/coordination/raft_state.cpp:135]: I am not sure if we need this log?

File: src/coordination/raft_state.cpp (hunk 1 above)
Comment (antoniofilipovic) [reply src/coordination/raft_state.cpp:174]: Seems like this was testing, so I think we should remove this, right?

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Looks good :muscle:

File: src/coordination/raft_state.cpp (hunk 1 above)
Comment (as51340) [reply src/coordination/raft_state.cpp:174]: yes

File: src/coordination/raft_state.cpp (hunk 2 above)
Comment (as51340) [reply src/coordination/raft_state.cpp:135]: we don't, was part of testing

**Review by antoniofilipovic — APPROVED** [review]

**Comment by as51340** [comment]
@kgolubic release note Adding coordinators is now managed by NuRaft instead of explicit saving inside Raft log.

**Pull request merged**